util/osuser/user.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package osuser implements OS user lookup. It's a wrapper around os/user that
+// works on non-cgo builds.
+package osuser
+
+import (
+	"context"
+	"errors"
+	"log"
+	"os/exec"
+	"os/user"
+	"runtime"
+	"strings"
+	"time"
+	"unicode/utf8"
+
+	"tailscale.com/version/distro"
+)
+
+// LookupUIDAndShell is like os/user.LookupId but handles a few edge cases like
```

Comment:
"And" makes this name a bit weird now. "LookupUID" really means "Lookup by UID" and the "by" was implicit. That worked before. But once you say "LookupUIDAndShell", you're not looking up _by_ the UID and Shell, and not looking up _by_ the shell, so it now reads like you're looking up the UID and looking up the shell, but we're not looking up the UID.

Maybe "osuser.LookupWithShellByUID" and "osuser.LookupWithShellByUsername"? Kinda verbose, but more clear?